image.go

```diff
@@ -150,7 +150,10 @@ func computeLayerMIMEType(what string, layerCompression archive.Compression) (om
 }
 
 // Extract the container's whole filesystem as if it were a single layer.
-func (i *containerImageRef) extractRootfs() (io.ReadCloser, chan error, error) {
+// Takes copierStripOpts as argument which allows caller to configure
+// preserve nature of setuid,setgid,sticky and extended attributes
+// on extracted rootfs.
+func (i *containerImageRef) extractRootfs(copierStripOpts copier.GetOptions) (io.ReadCloser, chan error, error) {
```

Comment:
Go ahead and define a `define.ExtractRootfsOptions` struct that only includes fields whose values are used.